sunpy/io/meta/fits_meta.py

```diff
@@ -0,0 +1,655 @@
+            self.schema = read_schema_file(schema_file)
+            these_required_keywords = self.schema[self.schema['required']]
+            required_keywords = vstack([required_keywords, these_required_keywords])
+        else:
+            self.schema = None
+
+        # fill the header with the required keywords
+        cards = []
+        for this_item in required_keywords:
+            allowed_values = this_item['allowed_values']
+            if allowed_values is not None and len(allowed_values) == 1:
+                cards.append(Card(keyword=this_item['keyword'], comment=this_item['description'], value=allowed_values[0]))
+            else:
+                cards.append(Card(keyword=this_item['keyword'], comment=this_item['description'], value=""))
+        super().__init__(cards=cards)
+
+    def validate(self):
+        """
+        Validate the keywords against the solarnet recommendations and a custom 
+        schema if provided.
+
+        The following checks are performed.
+          * Check that all required keywords are present
+          * Check that keywords that have limited allowed values are consistent with those.
+          * Check that keyword values are consistent with the defined data types
+          * Check that keywords that are quantities define the unit properly in the comment.
+          * Check that an observer position is provided.
+          * Check that the comment are standard.
+          * Check that all keywords have values.
+        
+          Returns
+          -------
+          result: bool
+            True if no issues were found.
+        """
+        if self.schema is not None:
+            self._validate_against_schema(self.schema)
+        # validate against solarnet schema
+        self._validate_against_schema(self.schema_solarnet)
+
+        # some special solarnet validations that cannot be included in the schema.
+        # the value of the keyword is a unit
+        for this_keyword in ['WAVEUNIT', 'BUNIT']:
+            self._check_unit(self.cards[this_keyword])
+        
+        # EXTNAME must not start with a space
+        if ['EXTNAME'] in self:
+            if not re.match(r'\w', self['EXTNAME']):
+                warn_user('EXTNAME must not start with a space.')
+
+        self.check_for_deprecated_keywords()
+
+        # TODO: check that an observer position is provided.
+
+    def _validate_against_schema(self, schema):
+        """Validate a set of header keywords against a schema.
+        
+        The following checks are performed.
+          * Check that all required keywords are present.
+          * Check that keywords that have limited allowed values are consistent with those.
+          * Check that keyword values are consistent with the defined data types.
+          * Check that keywords that are quantities define the unit properly in the comment.
+          * Check that the comment are standard.
+          * Check that all keywords have values.
```

Comment:
I think it would be better to separate out each of these steps into its own function and then have this method call each of these functions.